code/game/machinery/turrets/turrets.dm

```diff
@@ -0,0 +1,225 @@
+		return
+	
+	// Draw idle power from cell if the area has no regular power.
+	else if(stat & NOPOWER)
+		// See if the turret should lose power.
+		if(!internal_cell.check_charge(idle_power_usage * CELLRATE))
+			update_use_power(POWER_USE_OFF)
+		//	set_state(TURRET_DEPOWERED)
+			// TODO ditto
+			return
+		else
+			internal_cell.use(idle_power_usage * CELLRATE)
+
+	// Charge the cell if possible.
+	else if(can_charge_cell && !internal_cell.fully_charged())
+		var/power_to_give = active_power_usage * CELLRATE
+	//	var/multiplier = time_per_process() / SSobj.wait // This is used so when the turret is using SSfastprocess it doesn't get supercharged.
+		var/multiplier = SSfastprocess.wait / SSmachines.wait // TODO remove
+		internal_cell.give(power_to_give * multiplier)
+		update_use_power(POWER_USE_ACTIVE)
+	// Draw idle power from APC.
+	else
+		update_use_power(POWER_USE_IDLE)
+
+/obj/machinery/turret/proc/is_depowered()
+	return (stat & BROKEN) || !internal_cell || !internal_cell.check_charge(idle_power_usage * CELLRATE)
+
+/*
+/obj/machinery/turret/proc/time_per_process()
+	return speed_process ? SSfastprocess.wait : SSobj.wait
+*/
+
+
+/obj/machinery/turret/proc/process_turning()
+	var/distance_from_target_bearing = closer_angle_difference(current_bearing, target_bearing)
+	var/distance_this_step = min(calculate_turn_rate_per_process(), abs(distance_from_target_bearing)) 
+	var/adjustment = distance_this_step * SIGN(distance_from_target_bearing)
+
+	// This probably won't work if traverse is greater than 180.
+	// If someone smarter than me knows how to make that work then please tell me how.
+	if(!angle_within_traverse(current_bearing + adjustment))
+		adjustment = adjustment * -1
+
+	set_bearing(current_bearing + adjustment)
+
+/obj/machinery/turret/proc/calculate_turn_rate_per_process()
+	return turning_rate / (1 SECOND / SSfastprocess.wait)
+
+
+/obj/machinery/turret/proc/process_shooting()
+	if(installed_gun)
+		installed_gun.Fire(target, src)
+
+
+/obj/machinery/turret/update_icon()
+	if(!turret_stand)
+		turret_stand = image(icon, "turretCover")
+		turret_stand.layer = src.layer - 0.01
+		turret_stand.appearance_flags = KEEP_APART|RESET_TRANSFORM|TILE_BOUND|PIXEL_SCALE 
+		overlays += turret_stand
+	
+	if(!turret_ray)
+		turret_ray = image(icon, "turret_ray")
+		turret_ray.plane = EFFECTS_ABOVE_LIGHTING_PLANE // Should it go below lighting?
+		turret_ray.appearance_flags = KEEP_APART|RESET_COLOR|TILE_BOUND|PIXEL_SCALE
+		turret_ray.mouse_opacity = FALSE
+	
+		var/matrix/M = matrix(turret_ray.transform)
+		// Offset away from the parent, so that when the parent rotates, this rotates with it correctly.
+		M.Translate(0, (1 * WORLD_ICON_SIZE * 0.50)+4)
+		M.Scale(1, 4)
+		turret_ray.transform = M
+		overlays += turret_ray
+
+	// Changes the ray color based on state.
+	overlays -= turret_ray
+	turret_ray.color = ray_color
+	overlays += turret_ray
+
+	..()
+
+/obj/machinery/turret/proc/calculate_traverse()
+	if(traverse >= 360)
```

Comment:
If you just want to clamp it to a sane angle, you can do `traverse = traverse % 360` I believe.

Reply:
Nvm this is literally what SIMPLIFY_DEGREES does.